### PR TITLE
Converting detected plain text mentions into permalinks

### PR DESCRIPTION
**Pull request by Velin92**
Works perfectly now

### CODE REVIEW

**Comment by Velin92** [comment]
@pixlwave  maybe we should skip converting room alias mentions into permalinks, and just kip the default link behaviour, even if it doesn't work properly?

File: ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift [hunk 1]
```diff
@@ -320,3 +325,31 @@ protocol MentionBuilderProtocol {
     func handleUserMention(for attributedString: NSMutableAttributedString, in range: NSRange, url: URL, userID: String)
     func handleAllUsersMention(for attributedString: NSMutableAttributedString, in range: NSRange)
 }
+
+private struct TypedMatch {
```
Comment (pixlwave) [inline ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift:329]: Very nice 👌

File: UnitTests/Sources/AttributedStringBuilderTests.swift [hunk 2]
```diff
@@ -191,22 +192,25 @@ class AttributedStringBuilderTests: XCTestCase {
     func testUserIdLink() {
         let userId = "@user:matrix.org"
         let string = "The user is \(userId)."
-        checkLinkIn(attributedString: attributedStringBuilder.fromHTML(string), expectedLink: userId, expectedRuns: 3)
-        checkLinkIn(attributedString: attributedStringBuilder.fromPlain(string), expectedLink: userId, expectedRuns: 3)
+        let expectedLink = "\(permalinkBaseURL)/#/\(userId)"
```
Comment (pixlwave) [inline UnitTests/Sources/AttributedStringBuilderTests.swift:195]: Not an issue, but I'm curious why this one doesn't percent encode the `:` but the other 2 do.

**Review by pixlwave — APPROVED** [review]
Dude this is great 👏

**Pull request merged**